src/middleware/company.authentication.middleware.ts

```diff
@@ -40,11 +27,32 @@ export const companyAuthentication = async (req: Request, res: Response, next: N
       logger.infoRequest(req, `Invoking company authentication with (${ entityNumber }) present in session`);
       return authMiddleware(authMiddlewareConfig)(req, res, next);
     } else {
-      logger.errorRequest(req, "No entity number to authenticate against");
-      throw new Error("OE number not found in session");
+      logger.errorRequest(req, "No entity number to authenticate against -- redirecting to start of Journey: " + UPDATE_LANDING_URL);
+      return res.redirect(UPDATE_LANDING_URL);
     }
   } catch (err) {
     logger.errorRequest(req, err);
     next(err);
   }
 };
+
+async function processTransaction (entityNumber: string, returnURL: string, req: Request): Promise<[string, string]> {
+  const { transactionId } = req.params;
+
+  if (transactionId) {
+    const transactionResource = await getTransaction(req, transactionId);
+    const entityNumberTransaction = transactionResource.companyNumber;
+
+    if (!entityNumber || (entityNumberTransaction && entityNumber !== entityNumberTransaction)) {
+      entityNumber = entityNumberTransaction as string;
+      returnURL = req.originalUrl;
+    }
+    if (entityNumber && transactionResource.status === CLOSED_PENDING_PAYMENT) {
```

Comment:
Does this not mean that anyone can browse a closed transaction? As long as they have an entityNumber?

Reply:
Should they be re-directed to payments URL here?